packages/back-end/src/models/WebhookSecretModel.ts

```
@@ -47,11 +47,17 @@ export class WebhookSecretDataModel extends BaseClass {
     });
   }
 
-  public async getBackEndSecretsReplacer() {
+  public async getBackEndSecretsReplacer(origin: string) {
     const secrets = await this.getAll();
     const replacements: Record<string, string> = {};
     for (const secret of secrets) {
-      replacements[secret.key] = secret.value;
+      if (
+        !secret.allowedOrigins ||
+        !secret.allowedOrigins.length ||
+        secret.allowedOrigins.includes(origin)
+      ) {
+        replacements[secret.key] = secret.value;
```

Should this raise instead? If the replacement is not defined, the substitution will fail later but it'll be much harder to track.

Raising here would also give a chance to send back some meaningful error like "secret is restricted to origins: <list>, origin <caller origin> is not found in this list"